tests/unit/test_telemetry_handler.py

```diff
@@ -86,3 +106,59 @@ def test_ping_adzerk_i(self, mock_urlopen):
     def test_ping_adzerk_e(self, mock_urlopen):
         ping_adzerk('2,foo,bar')
         mock_urlopen.assert_called_once_with("https://e-10250.adzerk.net/e.gif?e=foo&s=bar")
+
+    @patch('google.cloud.logging')
+    @patch('logging.info')
+    def test_record_metrics_no_sampling(self, mock_logging, mock_google_logging):
+        os.environ["METRICS_SAMPLE_RATE"] = "0"
+        shim = make_encoded_shim(1713971071000)
+        record_metrics(f'2,{shim},bar', '2024-04-24T21:02:18.123456Z')
+        mock_logging.assert_not_called()
+
+    @patch('google.cloud.logging')
+    @patch('logging.info')
+    def test_record_metrics_sampling_misconfigured(self, mock_logging, mock_google_logging):
+        os.environ["METRICS_SAMPLE_RATE"] = "true"
+        shim = make_encoded_shim(1713971071000)
+        record_metrics(f'2,{shim},bar', '2024-04-24T21:02:18.123456Z')
+        mock_logging.assert_not_called()
+
+    @patch('google.cloud.logging')
+    @patch('logging.info')
+    def test_record_metrics_sample_rate_excluded(self, mock_logging, mock_google_logging):
+        # seed random to ensure random sample is excluded
+        random.seed(0)
+        os.environ["METRICS_SAMPLE_RATE"] = "500"
+        shim = make_encoded_shim(1713971071000)
+        record_metrics(f'2,{shim},bar', '2024-04-24T21:02:18.123456Z')
+        mock_logging.assert_not_called()
+
+    @patch('google.cloud.logging')
+    @patch('logging.info')
+    def test_record_metrics_shim_not_parsable(self, mock_logging, mock_google_logging):
+        # seed random to ensure random sample is excluded
+        random.seed(0)
+        os.environ["METRICS_SAMPLE_RATE"] = "900"
```

Comment:
Should this just be `os.environ["METRICS_SAMPLE_RATE"] = "1000"` without the need to set the random seed? IIUC logging shouldn't happen in this test regardless of the sample rate.